test/test_SensorFilter.py

```diff
@@ -25,6 +25,13 @@ def test_invalid_nonetype(self):
         expected_type = float
         self.assertFalse(SensorFilter.filter(reading, lowerbound, upperbound, expected_type))
 
+    def test_bad_type(self):
```

Comment:
If the reading is an int do we want to not update the field? Would anything break if the reading was an int instead of a float? Maybe you could just cast it to a float.